src/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/AggregationMinMaxHelpers.cs

```diff
@@ -40,18 +42,20 @@ private static T Reduce(IEnumerable<T> source, int sign)
         // Helper method to find the minimum element in the source.
         //
 
+        [return: MaybeNull]
         internal static T ReduceMin(IEnumerable<T> source)
         {
-            return Reduce(source, -1);
+            return Reduce(source, -1)!;
```

Comment:
Is this `!` needed even though we have a `MaybeNull` return attribute?

Reply:
Yes still need suppress compiler error, not sure if I should open an issue for Roslyn though